pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -962,6 +962,15 @@ func applyUpstreamTLSSettings(opts *buildClusterOpts, tls *networking.ClientTLSS
 
 	cluster := opts.cluster
 	proxy := opts.proxy
+
+	// Disable transport socket when cluster type is `Cluster_ORIGINAL_DST` and mtls is autoDetected
+	// We don't know whether headless service instance has sidecar injected or not.
+	if cluster.GetType() == apiv2.Cluster_ORIGINAL_DST {
```

Comment:
If we check the cluster type here, I think this part https://github.com/hzxuzhonghu/istio/blob/8f6ec6a5f9555a304bfab47c9bd50e4cc1611e41/pilot/pkg/networking/core/v1alpha3/cluster.go#L1072-L1091 is not needed anymore?